tools/c7n_gcp/c7n_gcp/filters/iampolicy.py

```diff
@@ -0,0 +1,49 @@
+from c7n.filters.core import ValueFilter
+
+from c7n.utils import local_session, type_schema
+
+
+class IamPolicyFilter(ValueFilter):
+    """Filters resources by its associated IAM Policy.
+
+    :example:
+
+    .. code-block :: yaml
+
+       policies:
+        - name: gcp-iam-policy
+          resource: gcp.kms-cryptokey
+          filters:
+            - type: iam-policy
+              key: "bindings[*].members"
+              op: intersect
+              value: ["allUsers", "allAuthenticatedUsers"]
+    """
+
+    schema = type_schema('iam-policy', rinherit=ValueFilter.schema)
+#     permissions = ('compute.instances.getEffectiveFirewalls',)
```

Comment:
Is there a single permission to get iam policies, or is it different permissions depending on the resource? If its just one, lets add it here.